### PR TITLE
refactor(cli): re-organize capability commands

### PR DESCRIPTION
**Pull request by Alive-Fish**
TASK: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/14150991
E2E TEST: https://github.com/OfficeDev/TeamsFx/actions/runs/2182128516

### CODE REVIEW

File: packages/cli/src/cmds/capability.ts [hunk 1]
```diff
@@ -20,23 +24,30 @@ import {
 } from "../telemetry/cliTelemetryEvents";
 import CLIUIInstance from "../userInteraction";
 import HelpParamGenerator from "../helpParamGenerator";
-import { automaticNpmInstallHandler } from "./preview/npmInstallHandler";
+import { automaticNpmInstallHandlerByObject } from "./preview/npmInstallHandler";
 
 abstract class CapabilityAddBase extends YargsCommand {
   abstract readonly yargsHelp: string;
+  abstract generateExculdeCaps(projectSettings: ProjectSettings | undefined): {
```
Comment (a1exwang) [inline packages/cli/src/cmds/capability.ts:31]: generateExcludeCaps typo?

File: packages/cli/src/cmds/capability.ts [hunk 2]
```diff
@@ -88,218 +100,53 @@ abstract class CapabilityAddBase extends YargsCommand {
...
-  public override modifyArguments(args: { [argName: string]: any }) {
-    CLIUIInstance.updatePresetAnswer("capabilities", args["capabilities"]);
-    delete args["capabilities"];
-    return args;
+  public override generateExculdeCaps(settings: ProjectSettings | undefined): {
+    frontend: boolean;
+    backend: boolean;
+    bot: boolean;
+  } {
+    return { frontend: ProjectSettingsHelper.includeFrontend(settings), backend: true, bot: true };
```
Comment (a1exwang) [inline packages/cli/src/cmds/capability.ts:114]: parameters are not used here?

File: packages/fx-core/src/plugins/solution/fx-solution/index.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ export * from "./solution";
 export * from "./constants";
 import "./v2/solution";
 import "./v3/solution";
+export * from "./question";
```
Comment (a1exwang) [inline packages/fx-core/src/plugins/solution/fx-solution/index.ts:5]: is this used?

Comment (Alive-Fish) [reply packages/fx-core/src/plugins/solution/fx-solution/index.ts:5]: Yes, use the `AzureSolutionQuestionNames` in `fx-solution`.

File: packages/cli/src/cmds/capability.ts (hunk 2 above)
Comment (Alive-Fish) [reply packages/cli/src/cmds/capability.ts:114]: I see it is `exclude` in the `automaticNpmInstallHandler` signature.

File: packages/cli/src/cmds/capability.ts (hunk 1 above)
Comment (Alive-Fish) [reply packages/cli/src/cmds/capability.ts]: Thanks, updated.

**Review by kuojianlu — DISMISSED** [review]

**Review by kuojianlu — APPROVED** [review]

**Review by chagong — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**